juriscraper/opinions/united_states/state/nd.py

```diff
@@ -50,11 +51,23 @@ def _process_html(self) -> None:
 
             for idx, txt in enumerate(raw_values[:5]):
                 if idx == 0:
-                    txt, _ = self.clean_name(txt)
+                    # Separate case name and citation if present
+                    match = re.match(
+                        r"^(.*?)(\s*((20\d{2}\sND\s\d+)|(1 \d\.N\.W3 d\d+)))?\s*$",
+                        txt,
+                    )
+                    if match:
+                        case_name = match.group(1).strip()
+                        citation = (
+                            match.group(2).strip() if match.group(2) else ""
+                        )
```

Comment:
this regex is going to fail for older cases 

for example `https://www.ndcourts.gov/supreme-court/opinions?topic=&author=&cit1=1999&citType=ND&cit2=1&searchQuery=&trialJudge=&pageSize=100&sortOrder=1` 

the citation is `1999 ND 1` and also doesnt account for NW2d pattern just the 3d.